modules/glslang/register_types.cpp

```diff
@@ -79,9 +79,10 @@ static Vector<uint8_t> _compile_shader_glsl(RenderingDevice::ShaderStage p_stage
 	glslang::TShader shader(stages[p_stage]);
 	CharString cs = p_source_code.ascii();
 	const char *cs_strings = cs.get_data();
+	const char *shader_name = "GLSL";
```

Comment:
The [spec](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/nonsemantic/NonSemantic.Shader.DebugInfo.100.asciidoc#DebugSource) wants this to be the full path to the source file
>File is an OpString holding the name of the source file including its full path.

Is it possible to pass the shader name to `_compile_shader_glsl` without breaking compatibility?

Reply:
I'm not sure if it will break compatibility. If that is a serious concern, could we only include the full path when  ``is_generate_spirv_debug_info_enabled()``?

IMO it would be nice to have access to the path here so we can print better error messages. 

CC @RandomShaper who likely knows better than I do

Reply:
I think we can. Neither `RenderingDevice::shader_set_compile_to_spirv_function` nor `_compile_shader_glsl()` are exposed to the user.

Regarding providing the path, given the GLSL is generated on the go, it will have to be the path to a temporary file. There are cases where users can provide .glsl files directly for compute shaders, and that one should be easier.

Regarding error messages (and also for the future, tracking which shader resource is being used in a given pipeline) so the error can identify the specific shader resource, that's tricky. I managed to do it for Godot 3, but it was too spartan for upstreaming. You have to book-keep the path from the user realm to the renderer realm, which requires some good plumbing. Also, the logic building the error string has to consider that some shaders just don't have an original path, which is the case of the graphics API shaders that don't come from user-level Godot shaders.

Reply:
I asked for clarification here: https://github.com/KhronosGroup/SPIRV-Registry/issues/204
It looks like providing an empty string is fine. The SPIRV binary generated by glslang contains the GLSL source, so there is no need for the shader to exist on the file system while debugging.